ng_model/src/lib.rs

```diff
@@ -4,6 +4,10 @@ pub use serde;
 pub use serde_json;
 pub use serde_with;
 
+const NONCE_MIN: u32 = 0;
```

Comment:
I don't think we need  constants for these, can create a type alias for u32:
```suggestion
type Nonce = u32;
```

And replace: `NONCE_MIN` with `Nonce::MIN`, 'NONCE_MAX` with `Nonce::MAX`, and `NONCE_SET_SPACE` is always going to be `Nonce::MAX`.